Makefile

```diff
@@ -50,18 +50,12 @@ GSUTIL ?=gsutil -m -q
 CODEGEN_PKG ?=./vendor/k8s.io/code-generator
 CODEGEN_HEADER_FILE ?=/dev/null
 
-# API_PACKAGES is a list of comma separated full api package refs (like golang import strings)
-API_PACKAGES ?=$(shell $(GO) list -json ./pkg/api/... | $(JQ) -sr 'map(select(.Name | test("^v[0-9]+.*$$"))) | reduce .[] as $$item ([]; . + [$$item.ImportPath]) | join(",")')$(if $(filter $(.SHELLSTATUS),0),,$(error failed to list api packages))
-ifeq "$(API_PACKAGES)" ""
-	$(error "API_PACKAGES can't be empty")
-endif
-NONREST_API_PACKAGES ?=$(shell $(GO) list -json ./pkg/scylla/api/... | $(JQ) -sr 'map(select(.Name | test("^v[0-9]+.*$$"))) | reduce .[] as $$item ([]; . + [$$item.ImportPath]) | join(",")')$(if $(filter $(.SHELLSTATUS),0),,$(error failed to list non-rest api packages))
-ifeq "$(NONREST_API_PACKAGES)" ""
-	$(error "NONREST_API_PACKAGES can't be empty")
-endif
+api_groups :=$(patsubst %/,%,$(wildcard ./pkg/api/*/))
+external_api_groups :=$(patsubst %/.,%,$(wildcard ./pkg/externalapi/*/.))
```

Comment:
I'm a bit confused with your intention here - you have the machinery set up for generation of CRDs, but there is no automation for updating and verification of third-party examples. So what is meant to happen if you update the sources of the monitoring API?

Reply:
yeah, that's a bit tricky and one predates the other. pkg/externalapi are our definition of prometheus compatible API in golang and generating CRDs just because that's what we do for APIs. Examples hold the deploy files from prometheus operator and other manifests. Unfortunately, there isn't a tool to generate golang structs from CRDs. So in the end those CRDs should be equivalent (or a subset) but with different version of generators it's hard to assess even the equivalency. Maybe we should just stop generating the CRDs in the pkg/externalapi but then it may be harder to asses a diff - wdyt?

Reply:
> Maybe we should just stop generating the CRDs in the pkg/externalapi 

What do you mean by that? 
I was thinking of just setting up the machinery to update the examples after the CRD are regenerated, the same way it's set up for the operator (i.e. update examples and verify examples). Maybe you could keep the prometheus-operator manfiests in examples, and generate all the `monitoring.coreos.com` CRDs in deploy and examples?

Reply:
Those generated CRDs for external API are not authoritative, we are not reconciling them - that's why I think we should deploy the definitions that come from the external repo that match what the external controller is reconciling.

If we have ever missed a field or resource or didn't choose to pick it (don't need it in our clients) we should still deploy the full/original definitions into the cluster where other apps can use it as they match the controller capabilities. Technically there can be some resources used only by the controller, so we should always use the original definitions.